### PR TITLE
similar resources carousel

### PR DESCRIPTION
**Pull request by gumaerc**
### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5878

### Description (What does it do?)
This PR adds two `ResourceCarousel`s to the new learning resource drawer based on the new "similar resources" APi endpoints. Both the opensearch and vector based results are displayed below the resource information.

### Screenshots (if appropriate):
image
image

### How can this be tested?
 - If you have Posthog set up locally, enable the `lr_drawer_v2` flag
 - If you don't have Posthog set up locally, you can force `drawerV2` to be `true` in `LearningResourceDrawer.tsx`
 - Spin up this branch of `mit-learn`
 - Generate vector embeddings by running `docker compose exec web python manage.py generate_embeddings --skip-contentfiles --recreate-collections --all`
 - Visit the search page at http://localhost:8062/search
 - View a variety of learning resources and ensure that the related resources carousels appear
 - Click the cards in the carousels to navigate to other learning resources and ensure that the drawer reloads with the new resource seamlessly

### CODE REVIEW

**Review by shanbady — CHANGES_REQUESTED** [review]

File: frontends/ol-components/src/components/LearningResourceExpanded/InfoSectionV2.tsx [hunk 1]
```diff
@@ -80,6 +80,7 @@ const IconContainer = styled.span({
   [theme.breakpoints.down("sm")]: {
     display: "none",
   },
+  zIndex: -1,
```
[the remark this answers is not in the document]
Comment (gumaerc) [reply frontends/ol-components/src/components/LearningResourceExpanded/InfoSectionV2.tsx:83]: I added this due to an issue with some elements of the drawer ending up "on top" of the sticky header in the drawer when you scroll down. I changed the approach here and instead set `zIndex: 1` on the header itself.

[the remark this answers is not in the document]
Comment (gumaerc) [reply frontends/ol-components/src/components/LearningResourceExpanded/InfoSectionV2.tsx:83]: Here's an example I captured on RC of the icons from `InfoSection` flowing over onto the header:
chrome_HBOmZykrsN

**Comment by gumaerc** [comment]
@shanbady This is ready for another look pending whatever @sovsey thinks we should do re: the empty space at the end of the carousel

File: frontends/ol-components/src/components/LearningResourceExpanded/InfoSectionV2.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (shanbady) [reply frontends/ol-components/src/components/LearningResourceExpanded/InfoSectionV2.tsx]: sounds good. just wanted to make sure it was something intentional and not left behind from previous testing.

**Review by shanbady — APPROVED** [review]
Looks good. Also worth mentioning @sovsey  the fix to resolve the performance issue on the vector results is up at https://github.com/mitodl/mit-learn/pull/1839 - if we have made a decision on the endpoint, we can use the vector based results once that pr is merged

**Comment by gumaerc** [comment]
@sovsey I'm going to merge this. If we decide we want to even out the amount of cards in the carousels as @shanbady suggested, I can put up a separate PR for that.

File: frontends/api/src/test-utils/urls.ts [hunk 2]
```diff
@@ -79,6 +79,11 @@ const learningResources = {
     `${API_BASE_URL}/api/v1/learning_resources/${params.id}/`,
   featured: (params?: Params<FeaturedApi, "featuredList">) =>
     `${API_BASE_URL}/api/v1/featured/${query(params)}`,
+  similar: (params: Params<LRApi, "learningResourcesSimilarList">) =>
+    `${API_BASE_URL}/api/v1/learning_resources/${params.id}/similar/`,
+  vectorSimilar: (
+    params: Params<LRApi, "learningResourcesVectorSimilarList">,
+  ) => `${API_BASE_URL}/api/v1/learning_resources/${params.id}/vector_similar/`,
```
[the remark this answers is not in the document]
Comment (sovsey) [reply frontends/api/src/test-utils/urls.ts:86]: Let's update so it evenly pages.

**Pull request merged**